services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -332,9 +316,7 @@ export const getNestedChildFields = (
 
   choices.forEach((choice: FieldChoice) => {
     // if choice is not selected and there are children
-    const isParentChoiceChecked = (id: string) =>
-      lastDatabaseValue?.some((autosave) => autosave.key === id);
-    if (choice.children && isParentChoiceChecked(choice.id)) {
```

Comment:
as mentioned above, tightened up the check for when we call `compileNestedFields(choice.children);`. This function runs any time a choice is checked or unchecked in a choicelist. It grabs all of the choices in a choicelist and checks to see if we need to clear out nested data from parent choices that are no longer selected